### DISCUSSION TITLE
Die 5590 ist gefreetzt

### DISCUSSION THREAD

**Discussion by freetz-ng-mod** [Freetz-NG/freetz-ng#534] [Allgemein]
![Bildschirmfoto vom 2022-05-18 18-39-18](https://user-images.githubusercontent.com/79326105/169098414-3a804286-54e2-451e-91b5-44350baf1594.png)
![Bildschirmfoto vom 2022-05-18 18-36-16](https://user-images.githubusercontent.com/79326105/169098424-81412f1b-a063-487f-8071-e23c27f6ef10.png)

**Comment by Himan2001** [comment]
Kannst Du noch einen weiteren Screenshot mit den entfernten Modulen noch posten ? (hardening ?, trustedd ?)

Mich interessiert speziell wie es mit dem trustedd hier rennt - oder muss der genauso wie bei der 5530 aktuell noch entfernt werden im developer mode ?

Rennen bei der Box LTE über einen Huawei USB-Stick als WAN-Option/Backup ?

**Comment by fda77** [comment]
Danke @freetz-ng-mod musstest du irgend etwas spezielles auswählen? Busybox version etc?
Hat push-firmware ohne zusätzliche Parameter funktioniert? Die Werte sind noch mit "?" markiert
Und teste(t) mal ob das flashen über freetz mit fit/uimg funktioniert (auf eigene Gefahr!!!), dazu diese Zeile löschen: https://github.com/Freetz-NG/freetz-ng/blob/master/make/mod/mod.mk#L30

**Comment by freetz-ng-mod** [comment]
@Himan2001 
![Bildschirmfoto vom 2022-05-18 20-27-46](https://user-images.githubusercontent.com/79326105/169118340-f9e7c602-d85b-40b0-87e2-cfc7fc0f5c24.png)
Da kannst sehen das ich nichts gelöscht habe oder groß angemacht habe

@fda77 
![Bildschirmfoto vom 2022-05-18 20-31-26](https://user-images.githubusercontent.com/79326105/169118775-702e70c1-5381-466d-9e9c-2caa0c7b5bcf.png)
Geht also immer noch nicht

push habe ich so gemacht
./push_firmware 5590_07.29.all-rev95450_release_freetz-ng-19392MO-81ae3a3c6.image -cmd ftp -map 0x40000000 -afu
und wo das ohne probs lief, habe ich über die AVM Gui das Image geflasht

Und noch was 
FREETZ_MODIFY_DSL_WARNING
FREETZ_MODIFY_DSL_SPECTRU
FREETZ_MODIFY_DSL_SETTINGS
Kannst bei der Box auch aus machen wie bei der 5530

**Reply by fda77** [reply]
Die dsl-dinge sind aus: 353d1e0f5

**Reply by fda77** [reply]
0x40000000 war schon default bei PF: 01af9ab9f

**Comment by freetz-ng-mod** [comment]
Danke.
Ich habe es einfach getestet, da es beim 6000 und 4060 auch so war. Aber nun sind wir uns sicher

**Reply by fda77** [reply]
Komisch beim flashen oben sind die 2  *3.image Dateien in der firmware, ich glaube die sollten dort nicht hin.
Die Meldung beim flashen ist komisch, war das vorher eine andere? Ich hab diese noch nie gesehen

**Comment by freetz-ng-mod** [comment]
Ist genauso wie beim 6000
https://github.com/Freetz-NG/freetz-ng/issues/465#issuecomment-1046472388

**Reply by fda77** [reply]
*3.image sind auch weg: a0dbdc11c

**Reply by freetz-ng-mod** [reply]
Dann baue ich mal ein neues Image und teste es noch mal

**Comment by fda77** [comment]
Dieses `CONFIG_ENVIRONMENT_PATH` wird wohl in der /etc/init.d/rc.conf gesetzt

**Reply by freetz-ng-mod** [reply]
Das heißt nun, was soll ich machen, um es zu testen. Denn noch habe ich das neu gebaute Image nicht geflash

**Reply by fda77** [reply]
Ich weiss nicht weshalb die Variable fehlt. Es kann sein dass das freetz-webif die nicht weitergibt, irgendwo gibt es eine Einschränkung was nach aussen "geleakt" wird. Müsste ich mal suchen, hab aber zuerst mal hunger

**Reply by fda77** [reply]
Oder doch, teste mal 
`echo $CONFIG_ENVIRONMENT_PATH`
einfach per ssh

**Reply by freetz-ng-mod** [reply]
echo $CONFIG_ENVIRONMENT_PATH
`/proc/sys/urlader`

**Comment by fda77** [comment]
Diese Fehlermeldung hab ich noch nie gesehen, der ähnliche Screenshot beim 6000 auch nicht
Vielleicht funktioniert flashen mit 7c2e427fd

**Comment by fda77** [comment]
Du könntest auch noch schauen ob 45495e948 funktioniert.

**Reply by freetz-ng-mod** [reply]
Also nun macht er es. Aber
![Bildschirmfoto vom 2022-05-19 05-36-56](https://user-images.githubusercontent.com/79326105/169200297-a57cd6a7-8869-43c5-be00-cbc05fde92a4.png)
Wie du sehen kannst, kommen keine Informationen mehr im 2 Fenster.

**Comment by fda77** [comment]
"nun macht er es" bedeutet dass die neue Firmware geflasht wurde? Und wieder für alle freigegeben werden kann?

Das flash-script hat scheinbar keine Ausgabe mehr. Wenn du die .image mit tar entpackst und var/install anschaust gibt es nur 2 Zeilen die etwas ausgeben könnten
```
/sbin/bootslotctl program_other /var/tmp/fit-image "${ver}-$(grep ^Build /var/content | cut -f2 -d=)" || exit 6
/sbin/bootslotctl "$_activate_action" || exit 6
```

Vor dem Erstellen dieser Box ist aber noch nicht bekannt ob es eine Ausgabe geben wird.
Man könnte sie bei "fit" immer abschalten, bei Fehlern wird aber vermutlich etwas angezeigt

**Comment by freetz-ng-mod** [comment]
Ja er schiebt das neu gebaute image auf die Box.

Lass mich das erst noch bei den anderen fit boxen die ich habe einmal testen bevor du es wieder frei gibst.

**Comment by fda77** [comment]
Wird bei den anderen auch so sein, da dieses "bootslotctl" genutzt wird.
Mit 4203dc8aa sollte die text-box nicht mehr leer sein.
Häng nochmal einen avm-info Screenshot an, ob emmc funktioniert

**Comment by freetz-ng-mod** [comment]
5590
![5590](https://user-images.githubusercontent.com/79326105/169362396-e657a7af-38e5-4b78-9f3e-a3f6b172d117.png)
Box Startet wann dann neu 
![55901](https://user-images.githubusercontent.com/79326105/169362913-d228bd1f-82f7-4d1c-a903-0ff069d99536.png)

**Reply by fda77** [reply]
Sieht soweit gut aus.
Flashen ist wieder aktiviert: cf93f1f22
Und das "keine Ausgabe" hab ich versucht von PRE zu einem "Hinweis" wie aus weiter unten zu ändern: e1389b793

**Comment by Himan2001** [comment]
Frage:

Kannst Du bitte das Image für die 5590 nochmal mit "hardening" und gleichen Optionen bauen und versuchen vom Stock AVM dann zu flashen ob das erfolgreich rennt, bzw. ob es beim trustedd dann zu Abstürzen kommt ? Falls ja, wäre die aktuelle Hardening Option mit dem Abschalten von "MyFreetz" und TR069 evtl. noch ein Problem bei der 5530 und 5590....?

Meine 5590 kommt auch noch die Tage. Aktuell kann ich bei gesetztem Hardening mit push_firmware nicht von der Stockfirmware
erstmalig freetzen. Das geht bei mir nur wenn ich den trustedd deaktiviere in Kombination mit entferntem TR069+MyFreetz Kram.

**Reply by freetz-ng-mod** [reply]
Das habe ich im 5530 ja schon mal gesagt ohne trustedd deaktiviere, geht nur das push der 7.21. Bei der 5590 brauch man es nicht.

**Reply by Himan2001** [reply]
...daher war ja meine Frage, mangels aktuell eigener 5590 noch, mal hardening zu aktivieren und zu schaun obs dann auf der 5590 weiterhin problemlos zu flashen ist und es auch zu keinem coredumps kommt :-)-  sofern es bei hardening dann zu problemen kommt kann ich zumindest bei TR069 und MyFreetz suchen, was ja damit deaktiviert wird ;-)

**Comment by freetz-ng-mod** [comment]
@fda77 
![Bildschirmfoto vom 2022-05-20 15-20-41](https://user-images.githubusercontent.com/79326105/169536739-7be533cb-dde5-4239-be42-97795723780d.png)
so schaut es gut aus ;-)

**Reply by fda77** [reply]
Funktioniert wie ich es gehofft hab... mit html hab ich nicht so viel Erfahrung
Wie baut sich das denn auf? Sieht man diese 2. PRE Box (ohne Ausgabe), sie verschwindet und dann geht es mit HINWEIS weiter? Oder ist das so schnell dass man es gar nicht wahrnimmt?

**Comment by freetz-ng-mod** [comment]
So wie das Bild ist, so schaut es aus. Es kommt überhaupt kein 2. PRE Box
Ich werde das mal bei der 7530AX machen. Da weiß ich ja das ich ein kleiner Fehler im speicher habe. Mal schauen, ob er das anzeigt

melde mich gleich 
![Bildschirmfoto vom 2022-05-20 15-55-53](https://user-images.githubusercontent.com/79326105/169543302-f7f9861a-5a6a-47b2-a893-1d09eda32090.png)

**Comment by fda77** [comment]
Wird schon passen. Mit https://github.com/Freetz-NG/freetz-ng/blob/master/make/mod/files/root/usr/lib/mww/do_update_handler.sh#L205 wird das HIDDEN der PRE erst nach Ausführung gesetzt, wenn leer. Wenn das so schnell ist dass man es nicht sieht ist prima.
Falls man Scripte im Browser deaktiviert hat bleibt halt die leere Box stehen, das ist dann so

**Comment by fda77** [comment]
Für edits wird man übrigens nicht benachrichtigt, hab den neuen Screenshot nur durch Zufall gesehen.
Die Erkennung ob es eine Ausgabe gab funktioniert jedenfalls.
Dein Flash hat einen kaputten Block, der durch einen aus der Reserve ersetzt wird. Das ist nur eine "info" und wie bei sd-karten oder ssds normal. Wenn es mal sehr viele sind sollte man sich um Ersatz kümmern (ich hab meine 7490 über 1000 mal geflasht)

**Reply by freetz-ng-mod** [reply]
Die 7530AX hat den Fehler, seit ich sie gekauft habe.

**Reply by Himan2001** [reply]
Aktuell kommt es bei mir nach wie vor zu einem Dauerproblem:

https://github.com/Freetz-NG/freetz-ng/issues/560
